### ISSUE TITLE
Goproxy creates files in your home directory as root

### ISSUE THREAD

**Issue by coryschwartz** [ipfs/testground#505]
We setup a bind mount to the default GOAPTH for the goproxy container
https://github.com/ipfs/testground/blob/master/pkg/build/golang/docker.go#L379

This container will drop files into ~/go/pkg as the root user.

Steps to reproduce:

```
1. Kill the goproxy container (if it's running)
2. mv ~/go ~/go.orig
3. ./testground daemon
4. ./testground run any test that will trigger a build
5. find ~/go/pkg -user root (there will be some)
```

I think this bind mount can just be removed, after all the Dockerfile already creates a volume for /go 

https://hub.docker.com/r/goproxy/goproxy/dockerfile

**Comment by Stebalien** [comment]
We wanted to share the cache with the user. We may need to mess with user IDs?

**Comment by raulk** [comment]
Exactly, the point was to share the host go mod cache with the build container. I think @Stebalien is right. Docker seems to start containers with `root` user by default (erm...), and we need to supply the API equivalent of the CLI `--user` flag with the current UID.

**Comment by Stebalien** [comment]
That should "just work". Ideally, we'd run under a _different_ user but that _doesn't_ look like it'll work without UID mapping shenanigans. Unfortunately, whoever implemented that thought that a system-wide static config file was a good idea so I'm not touching that with a 10 foot poll.

**Comment by Stebalien** [comment]
Actually, that won't "just work". We probably need to ensure that the right directories in the container are owned by that user.

**Comment by coryschwartz** [comment]
The PR I submitted is just to remove the bind mount.

I think sharing the user go pkg dir is ultimately unnecessary. The goproxy remains useful as a cache because the docker volume it creates at persists between container restarts.

Does this seem reasonable, or is there a compelling reason to keep the bind mount I haven't thought of?

**Comment by Stebalien** [comment]
It's probably fine. And yeah, creating files as root is strictly worse.

It does kind of suck that we'll have to re-download everything after a system prune, but that's not the end of the world.

**Comment by raulk** [comment]
This was fixed in #591.
